presto-main/src/test/java/io/prestosql/sql/planner/AbstractPredicatePushdownTest.java

```diff
@@ -440,6 +443,36 @@ public void testNonDeterministicPredicateNotPushedDown()
                                                                 ImmutableMap.of("CUST_KEY", "custkey"))))))));
     }
 
+    @Test
+    public void testNormalizeOuterJoinToInner()
+    {
+        Session disableJoinReordering = Session.builder(getQueryRunner().getDefaultSession())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, "NONE")
+                .build();
+
+        assertPlan(
+                "SELECT customer.name, lineitem.partkey " +
+                        "FROM lineitem " +
+                        "LEFT JOIN orders ON lineitem.orderkey = orders.orderkey " +
+                        "INNER JOIN customer ON orders.custkey = customer.custkey " +
+                        "WHERE customer.name IS NOT NULL",
+                disableJoinReordering,
+                anyTree(
+                        join(
+                                INNER,
+                                ImmutableList.of(equiJoinClause("o_custkey", "c_custkey")),
+                                anyTree(
+                                        join(
+                                                enableDynamicFiltering ? INNER : LEFT, // TODO (https://github.com/prestosql/presto/issues/2392) this should be INNER also when dynamic filtering is off
```

Comment:
make this test DF independent (or with DF disabled). No need to mix DF here

Reply:
This test is run with and without DF.
As you see, it *is* DF-dependent -- the actual results depend on whether DF is enabled.

In the future, eg when we do #2392, the assertion will be simplified to be always INNER, and this will look better.

Reply:
> As you see, it is DF-dependent -- the actual results depend on whether DF is enabled.

Right, but we don't need to run this test twice to make sure normalization works. Normalization part is completely independent of DFs

Reply:
So this normalization works independently from DF:

https://github.com/prestosql/presto/pull/3242/files#diff-3c91106969a4d5786ab8f7ef09d73b3bR462

and this normalization depends on DF:

https://github.com/prestosql/presto/pull/3242/files#diff-3c91106969a4d5786ab8f7ef09d73b3bR466

Reply:
I think we want to test `io.prestosql.sql.planner.optimizations.PredicatePushDown.Rewriter#tryNormalizeToOuterToInnerJoin`. It doesn't dependent on DFs

Reply:
Yes, the conversion LEFT -> INNER happens in `tryNormalizeToOuterToInnerJoin.
No, the conversion depends on DF, because DF adds filters which allow `tryNormalizeToOuterToInnerJoin` to do its job. With DF disabled, `tryNormalizeToOuterToInnerJoin` does not get any information it could act upon.

This will be alleviated once we figure out how to do https://github.com/prestosql/presto/issues/2392 without CPU hit.

Reply:
> tryNormalizeToOuterToInnerJoin` does not get any information it could act upon.

`tryNormalizeToOuterToInnerJoin` pre-dated DFs and worked without them too.